shared/src/main/scala/coop/rchain/shared/Iso.scala

```diff
@@ -0,0 +1,22 @@
+package coop.rchain.shared
+import shapeless.LabelledGeneric
+
+trait Iso[A, B] {
+  def to(a: A): B
+  def from(b: B): A
+}
+
+object Iso {
```

Comment:
I understand what `isomorphism` is doing in this abstract context, but I don't understand why we are using it? Is that in order to abstract from the context from "what" a TLS is created?

Reply:
Ok, I see, dependencies between modules. Why the shapeless implementaiton though?

Reply:
I can't move `Tls` class to `comm` because it is part of the configuration, and I can't use it due to cyclic dependencies in sbt, so I just created the same class which is converted implicitly with `Iso` (to avoid that ugly manual converting).